ntbs-service/Helpers/EpisodesExtensionMethods.cs

```diff
@@ -29,16 +29,20 @@ public static List<TreatmentPeriod> GroupEpisodesIntoPeriods(this IEnumerable<Tr
 
             foreach (var treatmentEvent in treatmentEvents.OrderForEpisodes())
             {
-                // If a transfer out event, make a new period just for this event
-                if (treatmentEvent.TreatmentEventType == TreatmentEventType.TransferOut)
-                {
-                    treatmentPeriods.Add(TreatmentPeriod.CreateTransferPeriod(treatmentEvent));
-                }
                 // If at the start of a new treatment period, make it and add the event
-                else if (currentTreatmentPeriod == null)
+                if (currentTreatmentPeriod == null)
                 {
-                    currentTreatmentPeriod = TreatmentPeriod.CreateTreatmentPeriod(periodNumber, treatmentEvent);
-                    periodNumber++;
+                    // If a transfer out event, make a special period just for this event
+                    if (treatmentEvent.TreatmentEventType == TreatmentEventType.TransferOut)
+                    {
+                        currentTreatmentPeriod = TreatmentPeriod.CreateTransferPeriod(treatmentEvent);
+                    }
+                    else
+                    {
+                        currentTreatmentPeriod = TreatmentPeriod.CreateTreatmentPeriod(periodNumber, treatmentEvent);
+                        periodNumber++;
+                    }
```

Comment:
You could put this in a single ternary expression:
```
currentTreatmentPeriod = treatmentEvent.TreatmentEventType == TreatmentEventType.TransferOut
    ? TreatmentPeriod.CreateTransferPeriod(treatmentEvent)
    : TreatmentPeriod.CreateTreatmentPeriod(periodNumber++, treatmentEvent);
```
but it's also definitely arguable that it's more confusing that way. I'm happy with either.

Reply:
It did occur to me when doing this but I thought it was too difficult to read, especially with the well hidden `periodNumber++` editing that value so I chose this approach, even though it is not perfect